Something went wrong error for card, no 'dataset_query' returned by API/card

**Describe the bug**
When loading a user-created question, the generic "Something went wrong" (+flaming server) error page is shown.
It is not clear why this happened or what the exact repro steps are. Some notes:

- User interface default language is set to German
- Instance was upgraded from v0.35.4 to v0.39.2, question created under v0.35.4
- The GET /api/card endpoint does not include a "dataset_query" object, unlike a normal response
- The question has a join to a saved question as per #12928 / #16353
- The question has filters on:
  - an expression in the saved question
  - a date field, where the filter includes a time
- The expression is an integer field divided by the 'count' metric calculated in the question

If it's not able to be reproduced, I would like a pointer on how to fix the question in the backend.
Though I'm not sure if the "dataset_query" object looks correct or what should be changed if it's wrong.


**Logs**

The browser (Firefox 88) console says:
```
Error: TypeError: c.dataset_query is null
```

The Metabase database entry for the affected card (note leading space).
```
> select dataset_query from report_card where id = 123;

 {"type":"query","query":{"source-table":"card__122","joins":[{"fields":"all","source-table":555,"condition":["=",["field-literal","person_id","type/Text"],["joined-field","persons",["field-id",16666]]],"alias":"persons"}],"filter":["and",["<",["datetime-field",["field-literal","date_seen","type/Date"],"minute"],"2021-05-01T12:30:00"],["=",["field-literal","keep_entry","type/Float"],1]]},"database":12}
```

When viewing the question, the Metabase logs say:

```
qtp432197459-6257 | 2021-06-08 07:51:13,041 | ERROR | metabase.models.interface | Unable to normalize:
 {:type "query",
 :query
 {:source-table "card__122",
  :joins
  [{:fields "all",
    :source-table 555,
    :condition ["=" ["field-literal" "person_id" "type/Text"] ["joined-field" "persons" ["field-id" 16666]]],
    :alias "persons"}],
  :filter
  ["and"
   ["<" ["datetime-field" ["field-literal" "date_seen" "type/Date"] "minute"] "2021-05-01T12:30:00"]
   ["=" ["field-literal" "keep_entry" "type/Float"] 1]]},
 :database 12}

clojure.lang.ExceptionInfo: Error normalizing query
        at metabase.mbql.normalize$fn__21188$fn__21189.invoke(normalize.cljc:795) ~[metabase.jar:?]
        at metabase.models.interface$maybe_normalize.invokeStatic(interface.clj:68) ~[metabase.jar:?]
        at metabase.models.interface$maybe_normalize.invoke(interface.clj:67) ~[metabase.jar:?]
        at metabase.models.interface$catch_normalization_exceptions$fn__24676.invoke(interface.clj:78) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at toucan.models$apply_type_fns$iter__24558__24562$fn__24563.invoke(models.clj:304) ~[metabase.jar:?]
        at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[metabase.jar:?]
        at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[metabase.jar:?]
        at clojure.lang.RT.seq(RT.java:535) ~[metabase.jar:?]
        at clojure.core$seq__5402.invokeStatic(core.clj:137) ~[metabase.jar:?]
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8146.invokeStatic(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8146.invoke(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13) ~[metabase.jar:?]
        at clojure.core$reduce.invokeStatic(core.clj:6828) ~[metabase.jar:?]
        at clojure.core$into.invokeStatic(core.clj:6895) ~[metabase.jar:?]
        at clojure.core$into.invoke(core.clj:6887) ~[metabase.jar:?]
        at toucan.models$apply_type_fns.invokeStatic(models.clj:302) ~[metabase.jar:?]
        at toucan.models$apply_type_fns.invoke(models.clj:299) ~[metabase.jar:?]
        at toucan.models$do_post_select.invokeStatic(models.clj:349) ~[metabase.jar:?]
        at toucan.models$do_post_select.invoke(models.clj:344) ~[metabase.jar:?]
        at toucan.db$do_post_select$iter__28532__28536$fn__28537.invoke(db.clj:373) ~[metabase.jar:?]
        at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[metabase.jar:?]
        at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[metabase.jar:?]
        at clojure.lang.RT.seq(RT.java:535) ~[metabase.jar:?]
        at clojure.lang.LazilyPersistentVector.create(LazilyPersistentVector.java:44) ~[metabase.jar:?]
        at clojure.core$vec.invokeStatic(core.clj:377) ~[metabase.jar:?]
        at clojure.core$vec.invoke(core.clj:367) ~[metabase.jar:?]
        at toucan.db$do_post_select.invokeStatic(db.clj:372) ~[metabase.jar:?]
        at toucan.db$do_post_select.invoke(db.clj:363) ~[metabase.jar:?]
        at toucan.db$simple_select.invokeStatic(db.clj:394) ~[metabase.jar:?]
        at toucan.db$simple_select.invoke(db.clj:383) ~[metabase.jar:?]
        at toucan.db$simple_select_one.invokeStatic(db.clj:420) ~[metabase.jar:?]
        at toucan.db$simple_select_one.invoke(db.clj:409) ~[metabase.jar:?]
        at toucan.db$select_one.invokeStatic(db.clj:627) ~[metabase.jar:?]
        at toucan.db$select_one.doInvoke(db.clj:620) ~[metabase.jar:?]
        at clojure.lang.RestFn.applyTo(RestFn.java:139) ~[metabase.jar:?]
        at clojure.lang.Var.applyTo(Var.java:705) ~[metabase.jar:?]
        at clojure.core$apply.invokeStatic(core.clj:671) ~[metabase.jar:?]
        at clojure.core$apply.invoke(core.clj:660) ~[metabase.jar:?]
        at toucan.models$invoke_model.invokeStatic(models.clj:380) ~[metabase.jar:?]
        at toucan.models$invoke_model.doInvoke(models.clj:368) ~[metabase.jar:?]
        at clojure.lang.RestFn.invoke(RestFn.java:445) ~[metabase.jar:?]
        at toucan.models$invoke_model.invokeStatic(models.clj:378) ~[metabase.jar:?]
        at toucan.models$invoke_model.invoke(models.clj:368) ~[metabase.jar:?]
        at clojure.lang.AFn.applyToHelper(AFn.java:156) ~[metabase.jar:?]
        at clojure.lang.RestFn.applyTo(RestFn.java:132) ~[metabase.jar:?]
        at clojure.core$apply.invokeStatic(core.clj:667) ~[metabase.jar:?]
        at clojure.core$apply.invoke(core.clj:660) ~[metabase.jar:?]
        at toucan.models$invoke_model_or_instance.invokeStatic(models.clj:395) ~[metabase.jar:?]
        at toucan.models$invoke_model_or_instance.doInvoke(models.clj:391) ~[metabase.jar:?]
        at clojure.lang.RestFn.invoke(RestFn.java:423) ~[metabase.jar:?]
        at metabase.models.card.CardInstance.invoke(card.clj:25) ~[metabase.jar:?]
        at metabase.api.card$fn__65802.invokeStatic(card.clj:162) ~[metabase.jar:?]
        at metabase.api.card$fn__65802.invoke(card.clj:159) ~[metabase.jar:?]
        at compojure.core$wrap_response$fn__12050.invoke(core.clj:160) ~[metabase.jar:?]
        at compojure.core$wrap_route_middleware$fn__12034.invoke(core.clj:132) ~[metabase.jar:?]
        at compojure.core$wrap_route_info$fn__12039.invoke(core.clj:139) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:151) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:153) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at metabase.server.middleware.auth$enforce_authentication$fn__56336.invoke(auth.clj:14) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at compojure.core$make_context$handler__12090.invoke(core.clj:287) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:296) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:199) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at clojure.lang.AFn.applyToHelper(AFn.java:160) ~[metabase.jar:?]
        at clojure.lang.AFn.applyTo(AFn.java:144) ~[metabase.jar:?]
        at clojure.core$apply.invokeStatic(core.clj:665) ~[metabase.jar:?]
        at clojure.core$apply.invoke(core.clj:660) ~[metabase.jar:?]
        at metabase.server.routes$fn__83020$fn__83021.doInvoke(routes.clj:57) ~[metabase.jar:?]
        at clojure.lang.RestFn.invoke(RestFn.java:436) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at compojure.core$make_context$handler__12090.invoke(core.clj:287) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:296) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$wrap_route_matches$fn__12043.invoke(core.clj:152) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:199) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063$respond_SINGLEQUOTE___12064.invoke(core.clj:197) ~[metabase.jar:?]
        at compojure.core$make_context$fn__12092.invoke(core.clj:297) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062$f__12063.invoke(core.clj:198) ~[metabase.jar:?]
        at compojure.core$routes$fn__12062.invoke(core.clj:200) ~[metabase.jar:?]
        at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__81060.invoke(exceptions.clj:98) ~[metabase.jar:?]
        at metabase.server.middleware.exceptions$catch_api_exceptions$fn__81057.invoke(exceptions.clj:86) ~[metabase.jar:?]
        at metabase.server.middleware.log$log_api_call$fn__82936$fn__82937.invoke(log.clj:195) ~[metabase.jar:?]
        at toucan.db$_do_with_call_counting.invokeStatic(db.clj:216) ~[metabase.jar:?]
        at toucan.db$_do_with_call_counting.invoke(db.clj:209) ~[metabase.jar:?]
        at metabase.server.middleware.log$log_api_call$fn__82936.invoke(log.clj:189) ~[metabase.jar:?]
        at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__82572.invoke(browser_cookie.clj:30) ~[metabase.jar:?]
        at metabase.server.middleware.security$add_security_headers$fn__81022.invoke(security.clj:142) ~[metabase.jar:?]
        at metabase.server.middleware.json$wrap_json_body$fn__82717.invoke(json.clj:62) ~[metabase.jar:?]
        at metabase.server.middleware.json$wrap_streamed_json_response$fn__82735.invoke(json.clj:98) ~[metabase.jar:?]
        at ring.middleware.keyword_params$wrap_keyword_params$fn__83277.invoke(keyword_params.clj:55) ~[metabase.jar:?]
        at ring.middleware.params$wrap_params$fn__83293.invoke(params.clj:69) ~[metabase.jar:?]
        at metabase.server.middleware.misc$maybe_set_site_url$fn__35809.invoke(misc.clj:58) ~[metabase.jar:?]
        at metabase.server.middleware.session$bind_current_user$fn__43391$fn__43392.invoke(session.clj:248) ~[metabase.jar:?]
        at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:229) ~[metabase.jar:?]
        at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:221) ~[metabase.jar:?]
        at metabase.server.middleware.session$bind_current_user$fn__43391.invoke(session.clj:247) ~[metabase.jar:?]
        at metabase.server.middleware.session$wrap_current_user_info$fn__43378.invoke(session.clj:207) ~[metabase.jar:?]
        at metabase.server.middleware.session$wrap_session_id$fn__43364.invoke(session.clj:153) ~[metabase.jar:?]
        at metabase.server.middleware.auth$wrap_api_key$fn__56344.invoke(auth.clj:27) ~[metabase.jar:?]
        at ring.middleware.cookies$wrap_cookies$fn__83197.invoke(cookies.clj:216) ~[metabase.jar:?]
        at metabase.server.middleware.misc$add_content_type$fn__35792.invoke(misc.clj:27) ~[metabase.jar:?]
        at metabase.server.middleware.misc$disable_streaming_buffering$fn__35817.invoke(misc.clj:75) ~[metabase.jar:?]
        at ring.middleware.gzip$wrap_gzip$fn__83239.invoke(gzip.clj:86) ~[metabase.jar:?]
        at metabase.server.middleware.misc$bind_request$fn__35820.invoke(misc.clj:92) ~[metabase.jar:?]
        at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__82953.invoke(ssl.clj:48) ~[metabase.jar:?]
        at metabase.server$async_proxy_handler$fn__82516.invoke(server.clj:71) ~[metabase.jar:?]
        at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source) ~[metabase.jar:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[metabase.jar:?]
        at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[metabase.jar:?]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383) ~[metabase.jar:?]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556) [metabase.jar:?]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375) [metabase.jar:?]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273) [metabase.jar:?]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [metabase.jar:?]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [metabase.jar:?]
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773) [metabase.jar:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905) [metabase.jar:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: clojure.lang.ExceptionInfo: Error canonicalizing query
        at metabase.mbql.normalize$canonicalize.invokeStatic(normalize.cljc:677) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize.invoke(normalize.cljc:665) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__21188$fn__21189.invoke(normalize.cljc:793) ~[metabase.jar:?]
        ... 201 more
Caused by: clojure.lang.ExceptionInfo: Invalid MBQL clause
        at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__21033.invoke(normalize.cljc:555) ~[metabase.jar:?]
        at clojure.walk$prewalk.invokeStatic(walk.clj:65) ~[metabase.jar:?]
        at clojure.walk$prewalk.invoke(walk.clj:61) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538) ~[metabase.jar:?]
        at medley.core$map_vals$fn__10074$fn__10075.invoke(core.cljc:110) ~[metabase.jar:?]
        at clojure.core$fn__8429$fn__8431.invoke(core.clj:6840) ~[metabase.jar:?]
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8140.invokeStatic(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8140.invoke(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13) ~[metabase.jar:?]
        at clojure.core$reduce.invokeStatic(core.clj:6828) ~[metabase.jar:?]
        at clojure.core$fn__8429.invokeStatic(core.clj:6830) ~[metabase.jar:?]
        at clojure.core$fn__8429.invoke(core.clj:6830) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8167$G__8162__8176.invoke(protocols.clj:175) ~[metabase.jar:?]
        at clojure.core$reduce_kv.invokeStatic(core.clj:6856) ~[metabase.jar:?]
        at clojure.core$reduce_kv.invoke(core.clj:6847) ~[metabase.jar:?]
        at medley.core$reduce_map.invokeStatic(core.cljc:84) ~[metabase.jar:?]
        at medley.core$reduce_map.invoke(core.cljc:81) ~[metabase.jar:?]
        at medley.core$map_vals.invokeStatic(core.cljc:110) ~[metabase.jar:?]
        at medley.core$map_vals.invoke(core.cljc:105) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__21033.invoke(normalize.cljc:545) ~[metabase.jar:?]
        at clojure.walk$prewalk.invokeStatic(walk.clj:65) ~[metabase.jar:?]
        at clojure.walk$prewalk.invoke(walk.clj:61) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at clojure.core$update.invokeStatic(core.clj:6196) ~[metabase.jar:?]
        at clojure.core$update.invoke(core.clj:6188) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize.invokeStatic(normalize.cljc:672) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize.invoke(normalize.cljc:665) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__21188$fn__21189.invoke(normalize.cljc:793) ~[metabase.jar:?]
        ... 201 more
Caused by: clojure.lang.ExceptionInfo: Output of update-field-options does not match schema:

           [nil nil (named (not ("Invalid :temporal-unit for the specified :base-type." a-clojure.lang.PersistentArrayMap)) "options")]


        at metabase.mbql.util$fn__20573$update_field_options__20578.doInvoke(util.cljc:581) ~[metabase.jar:?]
        at clojure.lang.RestFn.applyTo(RestFn.java:142) ~[metabase.jar:?]
        at clojure.core$apply.invokeStatic(core.clj:669) ~[metabase.jar:?]
        at clojure.core$apply.invoke(core.clj:660) ~[metabase.jar:?]
        at metabase.mbql.util$assoc_field_options.invokeStatic(util.cljc:590) ~[metabase.jar:?]
        at metabase.mbql.util$assoc_field_options.doInvoke(util.cljc:587) ~[metabase.jar:?]
        at clojure.lang.RestFn.invoke(RestFn.java:439) ~[metabase.jar:?]
        at metabase.mbql.util$with_temporal_unit.invokeStatic(util.cljc:595) ~[metabase.jar:?]
        at metabase.mbql.util$with_temporal_unit.invoke(util.cljc:592) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__20861.invokeStatic(normalize.cljc:404) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__20861.invoke(normalize.cljc:398) ~[metabase.jar:?]
        at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_implicit_field_id.invokeStatic(normalize.cljc:363) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_implicit_field_id.invoke(normalize.cljc:360) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_simple_filter_clause.invokeStatic(normalize.cljc:455) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_simple_filter_clause.invoke(normalize.cljc:452) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__20918$fn__20919.invoke(normalize.cljc:464) ~[metabase.jar:?]
        at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[metabase.jar:?]
        at clojure.core$map$fn__5866.invoke(core.clj:2753) ~[metabase.jar:?]
        at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[metabase.jar:?]
        at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[metabase.jar:?]
        at clojure.lang.RT.seq(RT.java:535) ~[metabase.jar:?]
        at clojure.core$seq__5402.invokeStatic(core.clj:137) ~[metabase.jar:?]
        at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8146.invokeStatic(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8146.invoke(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13) ~[metabase.jar:?]
        at clojure.core$reduce.invokeStatic(core.clj:6828) ~[metabase.jar:?]
        at clojure.core$into.invokeStatic(core.clj:6895) ~[metabase.jar:?]
        at clojure.core$into.invoke(core.clj:6887) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_compound_filter_clause.invokeStatic(normalize.cljc:425) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_compound_filter_clause.invoke(normalize.cljc:423) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__20888$fn__20889.invoke(normalize.cljc:433) ~[metabase.jar:?]
        at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__21033.invoke(normalize.cljc:552) ~[metabase.jar:?]
        at clojure.walk$prewalk.invokeStatic(walk.clj:65) ~[metabase.jar:?]
        at clojure.walk$prewalk.invoke(walk.clj:61) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538) ~[metabase.jar:?]
        at medley.core$map_vals$fn__10074$fn__10075.invoke(core.cljc:110) ~[metabase.jar:?]
        at clojure.core$fn__8429$fn__8431.invoke(core.clj:6840) ~[metabase.jar:?]
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8140.invokeStatic(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8140.invoke(protocols.clj:75) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13) ~[metabase.jar:?]
        at clojure.core$reduce.invokeStatic(core.clj:6828) ~[metabase.jar:?]
        at clojure.core$fn__8429.invokeStatic(core.clj:6830) ~[metabase.jar:?]
        at clojure.core$fn__8429.invoke(core.clj:6830) ~[metabase.jar:?]
        at clojure.core.protocols$fn__8167$G__8162__8176.invoke(protocols.clj:175) ~[metabase.jar:?]
        at clojure.core$reduce_kv.invokeStatic(core.clj:6856) ~[metabase.jar:?]
        at clojure.core$reduce_kv.invoke(core.clj:6847) ~[metabase.jar:?]
        at medley.core$reduce_map.invokeStatic(core.cljc:84) ~[metabase.jar:?]
        at medley.core$reduce_map.invoke(core.cljc:81) ~[metabase.jar:?]
        at medley.core$map_vals.invokeStatic(core.cljc:110) ~[metabase.jar:?]
        at medley.core$map_vals.invoke(core.cljc:105) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__21033.invoke(normalize.cljc:545) ~[metabase.jar:?]
        at clojure.walk$prewalk.invokeStatic(walk.clj:65) ~[metabase.jar:?]
        at clojure.walk$prewalk.invoke(walk.clj:61) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at clojure.core$update.invokeStatic(core.clj:6196) ~[metabase.jar:?]
        at clojure.core$update.invoke(core.clj:6188) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize.invokeStatic(normalize.cljc:672) ~[metabase.jar:?]
        at metabase.mbql.normalize$canonicalize.invoke(normalize.cljc:665) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at clojure.core$comp$fn__5807.invoke(core.clj:2569) ~[metabase.jar:?]
        at metabase.mbql.normalize$fn__21188$fn__21189.invoke(normalize.cljc:793) ~[metabase.jar:?]
        ... 201 more
```


**To Reproduce**
Not sure - have not found exactly how to reproduce it with the Sample Database.

A similar question complains about having a time filter on a date field, but no "something went wrong"
```
Unsupported temporal bucketing: You can't bucket a :type/Date Field by :minute.
```

Maybe it is about the combination of metric + expression + saved question + filters?


**Expected behavior**
The question should load without error, or at least an error that the user can rectify.
The "something went wrong" page does not show any of the normal Question UI controls.

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, Safari 11.1, …) Firefox 88
- Your operating system: (e.x. OS X 10.10, Windows 10.1809, Ubuntu 16.04, …) CentOS 7
- Your databases: (e.x. MySQL, Postgres, MongoDB, …) PostgreSQL 10
- Metabase version: (e.x. 0.19.3) Metabase v0.39.2
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Jar-file on Ubuntu, …) CentOS 7
- Metabase internal database: (e.x. H2 (default), Postgres or MySQL) PostgreSQL 10

**Severity**
Blocking with respect to not being able to view or fix this particular question

Hi @lindsay-stevens 
Sorry for the delay, was a little tricky to figure out what was happening.

So the problem comes from the a regression since 0.39, which is giving problems for the function that expands the date-column, so it gets minutes too, so it can be filtered by minutes. But this seems wrong to allow hour/minute filtering on dates.
The core error is `Invalid :temporal-unit for the specified :base-type`

**Workaround:**
Expecting you only have one question causing problems, then just manually edit the `report_card.dataset_query` and change this:
... `["<",["datetime-field",["field-literal","date_seen","type/Date"],"minute"],"2021-05-01T12:30:00"]]` ...
To this instead:
... `["<",["field-literal","date_seen","type/Date"],"2021-05-01"]]` ...
If you have a lot questions like this, then I'll figure out a safe way of doing it for the entire table, but let me know.

**Reproduce:**
On 0.38.4
1. Native query > Sample Dataset > save as "Q1"
 `select 'Gadget' as mycat, '2021-06-09'::date as mydate`
2. Simple question > Saved Questions > "Q1" - Filter "MYDATE" before 2021-06-11 **and add time** - save as "Q2"
![image](https://user-images.githubusercontent.com/1447303/121725403-84416100-cae9-11eb-9243-b15764ae0b23.png)
3. Upgrade to 0.39+ and go to "Q2" - shows the full screen "Something went wrong" (not the popup) without the possibility to view/edit the question
<details><summary>Full stacktrace</summary>

```
2021-06-11 19:26:20,727 DEBUG middleware.log :: GET /api/user/current 200 4.4 ms (3 DB calls) App DB connections: 0/7 Jetty threads: 4/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
2021-06-11 19:26:20,825 DEBUG middleware.log :: GET /api/session/properties 200 28.6 ms (2 DB calls) App DB connections: 0/7 Jetty threads: 4/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
2021-06-11 19:26:21,024 DEBUG middleware.log :: GET /api/database 200 21.1 ms (3 DB calls) App DB connections: 0/7 Jetty threads: 4/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
2021-06-11 19:26:21,138 ERROR mbql.normalize :: Invalid clause: [:< [:datetime-field [:field-literal MYDATE :type/Date] :minute] 2021-06-11T12:30:00]
2021-06-11 19:26:21,214 ERROR models.interface :: Unable to normalize: 
 {:type "query",
 :query
 {:source-table "card__653",
  :filter ["<" ["datetime-field" ["field-literal" "MYDATE" "type/Date"] "minute"] "2021-06-11T12:30:00"]},
 :database 4}

clojure.lang.ExceptionInfo: Error normalizing query {:query {:type "query", :query {:source-table "card__653", :filter ["<" ["datetime-field" ["field-literal" "MYDATE" "type/Date"] "minute"] "2021-06-11T12:30:00"]}, :database 4}}
	at metabase.mbql.normalize$fn__19679$fn__19680.invoke(normalize.cljc:795)
	at metabase.models.interface$maybe_normalize.invokeStatic(interface.clj:68)
	at metabase.models.interface$maybe_normalize.invoke(interface.clj:67)
	at metabase.models.interface$catch_normalization_exceptions$fn__23860.invoke(interface.clj:78)
	at clojure.core$comp$fn__5807.invoke(core.clj:2569)
	at toucan.models$apply_type_fns$iter__23742__23746$fn__23747.invoke(models.clj:304)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24)
	at clojure.core.protocols$fn__8146.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8146.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6828)
	at clojure.core$into.invokeStatic(core.clj:6895)
	at clojure.core$into.invoke(core.clj:6887)
	at toucan.models$apply_type_fns.invokeStatic(models.clj:302)
	at toucan.models$apply_type_fns.invoke(models.clj:299)
	at toucan.models$do_post_select.invokeStatic(models.clj:349)
	at toucan.models$do_post_select.invoke(models.clj:344)
	at toucan.db$do_post_select$iter__28555__28559$fn__28560.invoke(db.clj:373)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.lang.LazilyPersistentVector.create(LazilyPersistentVector.java:44)
	at clojure.core$vec.invokeStatic(core.clj:377)
	at clojure.core$vec.invoke(core.clj:367)
	at toucan.db$do_post_select.invokeStatic(db.clj:372)
	at toucan.db$do_post_select.invoke(db.clj:363)
	at toucan.db$simple_select.invokeStatic(db.clj:394)
	at toucan.db$simple_select.invoke(db.clj:383)
	at toucan.db$simple_select_one.invokeStatic(db.clj:420)
	at toucan.db$simple_select_one.invoke(db.clj:409)
	at toucan.db$select_one.invokeStatic(db.clj:627)
	at toucan.db$select_one.doInvoke(db.clj:620)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:671)
	at clojure.core$apply.invoke(core.clj:660)
	at toucan.models$invoke_model.invokeStatic(models.clj:380)
	at toucan.models$invoke_model.doInvoke(models.clj:368)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at toucan.models$invoke_model.invokeStatic(models.clj:378)
	at toucan.models$invoke_model.invoke(models.clj:368)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at toucan.models$invoke_model_or_instance.invokeStatic(models.clj:395)
	at toucan.models$invoke_model_or_instance.doInvoke(models.clj:391)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at metabase.models.card.CardInstance.invoke(card.clj:25)
	at metabase.api.card$fn__56160.invokeStatic(card.clj:162)
	at metabase.api.card$fn__56160.invoke(card.clj:159)
	at compojure.core$wrap_response$fn__1996.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__1980.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__1985.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:151)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:153)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__75910.invoke(auth.clj:14)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008.invoke(core.clj:200)
	at compojure.core$make_context$handler__2036.invoke(core.clj:287)
	at compojure.core$make_context$fn__2038.invoke(core.clj:296)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$make_context$fn__2038.invoke(core.clj:297)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$make_context$fn__2038.invoke(core.clj:297)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$make_context$fn__2038.invoke(core.clj:297)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at metabase.api.routes$fn__76032$fn__76034.invoke(routes.clj:70)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.server.routes$fn__78303$fn__78304.doInvoke(routes.clj:57)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008.invoke(core.clj:200)
	at compojure.core$make_context$handler__2036.invoke(core.clj:287)
	at compojure.core$make_context$fn__2038.invoke(core.clj:296)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__1989.invoke(core.clj:152)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008$f__2009$respond_SINGLEQUOTE___2010.invoke(core.clj:197)
	at metabase.server.routes$fn__78291$fn__78293.invoke(routes.clj:41)
	at compojure.core$routes$fn__2008$f__2009.invoke(core.clj:198)
	at compojure.core$routes$fn__2008.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__76023.invoke(exceptions.clj:98)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__76020.invoke(exceptions.clj:86)
	at metabase.server.middleware.log$log_api_call$fn__78219$fn__78220.invoke(log.clj:195)
	at toucan.db$_do_with_call_counting.invokeStatic(db.clj:216)
	at toucan.db$_do_with_call_counting.invoke(db.clj:209)
	at metabase.server.middleware.log$log_api_call$fn__78219.invoke(log.clj:189)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__77855.invoke(browser_cookie.clj:30)
	at metabase.server.middleware.security$add_security_headers$fn__75985.invoke(security.clj:142)
	at metabase.server.middleware.json$wrap_json_body$fn__78000.invoke(json.clj:62)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__78018.invoke(json.clj:98)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__78560.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__78576.invoke(params.clj:69)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__35891.invoke(misc.clj:58)
	at metabase.server.middleware.session$bind_current_user$fn__42607$fn__42608.invoke(session.clj:257)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:238)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:230)
	at metabase.server.middleware.session$bind_current_user$fn__42607.invoke(session.clj:256)
	at metabase.server.middleware.session$wrap_current_user_info$fn__42594.invoke(session.clj:216)
	at metabase.server.middleware.session$wrap_session_id$fn__42580.invoke(session.clj:162)
	at metabase.server.middleware.auth$wrap_api_key$fn__75918.invoke(auth.clj:27)
	at ring.middleware.cookies$wrap_cookies$fn__78480.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__35874.invoke(misc.clj:27)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__35899.invoke(misc.clj:75)
	at ring.middleware.gzip$wrap_gzip$fn__78522.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__35902.invoke(misc.clj:92)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__78236.invoke(ssl.clj:38)
	at metabase.server$async_proxy_handler$fn__77799.invoke(server.clj:71)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: clojure.lang.ExceptionInfo: Error canonicalizing query {:query {:source-table "card__653", :filter [:< [:datetime-field [:field-literal "MYDATE" :type/Date] :minute] "2021-06-11T12:30:00"]}}
	at metabase.mbql.normalize$canonicalize.invokeStatic(normalize.cljc:677)
	at metabase.mbql.normalize$canonicalize.invoke(normalize.cljc:665)
	at clojure.core$comp$fn__5807.invoke(core.clj:2569)
	at clojure.core$comp$fn__5807.invoke(core.clj:2569)
	at metabase.mbql.normalize$fn__19679$fn__19680.invoke(normalize.cljc:793)
	... 190 more
Caused by: clojure.lang.ExceptionInfo: Invalid MBQL clause {:clause [:< [:datetime-field [:field-literal "MYDATE" :type/Date] :minute] "2021-06-11T12:30:00"]}
	at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__19524.invoke(normalize.cljc:555)
	at clojure.walk$prewalk.invokeStatic(walk.clj:65)
	at clojure.walk$prewalk.invoke(walk.clj:61)
	at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541)
	at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538)
	at medley.core$map_vals$fn__1779$fn__1780.invoke(core.cljc:110)
	at clojure.lang.PersistentArrayMap.kvreduce(PersistentArrayMap.java:377)
	at clojure.core$fn__8437.invokeStatic(core.clj:6845)
	at clojure.core$fn__8437.invoke(core.clj:6830)
	at clojure.core.protocols$fn__8167$G__8162__8176.invoke(protocols.clj:175)
	at clojure.core$reduce_kv.invokeStatic(core.clj:6856)
	at clojure.core$reduce_kv.invoke(core.clj:6847)
	at medley.core$reduce_map.invokeStatic(core.cljc:84)
	at medley.core$reduce_map.invoke(core.cljc:81)
	at medley.core$map_vals.invokeStatic(core.cljc:110)
	at medley.core$map_vals.invoke(core.cljc:105)
	at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__19524.invoke(normalize.cljc:545)
	at clojure.walk$prewalk.invokeStatic(walk.clj:65)
	at clojure.walk$prewalk.invoke(walk.clj:61)
	at metabase.mbql.normalize$canonicalize_mbql_clauses.invokeStatic(normalize.cljc:541)
	at metabase.mbql.normalize$canonicalize_mbql_clauses.invoke(normalize.cljc:538)
	at clojure.core$comp$fn__5807.invoke(core.clj:2569)
	at clojure.core$update.invokeStatic(core.clj:6196)
	at clojure.core$update.invoke(core.clj:6188)
	at metabase.mbql.normalize$canonicalize.invokeStatic(normalize.cljc:672)
	... 194 more
Caused by: clojure.lang.ExceptionInfo: Output of update-field-options does not match schema: 

	   [nil nil (named (not ("Invalid :temporal-unit for the specified :base-type." a-clojure.lang.PersistentArrayMap)) "options")]  

 {:type :schema.core/error, :schema (constrained [(one (eq :field) :field) (one (cond-pre (constrained Int "positive integer") (constrained Str "Non-blank string")) "id-or-name") (one (maybe (recursive (var metabase.mbql.schema/FieldOptions))) "options")] ":field clauses using a string field name must specify :base-type."), :value [:field "MYDATE" {:base-type :type/Date, :temporal-unit :minute}], :error [nil nil (named (not ("Invalid :temporal-unit for the specified :base-type." a-clojure.lang.PersistentArrayMap)) "options")]}
	at metabase.mbql.util$fn__19064$update_field_options__19069.doInvoke(util.cljc:585)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.mbql.util$assoc_field_options.invokeStatic(util.cljc:594)
	at metabase.mbql.util$assoc_field_options.doInvoke(util.cljc:591)
	at clojure.lang.RestFn.invoke(RestFn.java:439)
	at metabase.mbql.util$with_temporal_unit.invokeStatic(util.cljc:599)
	at metabase.mbql.util$with_temporal_unit.invoke(util.cljc:596)
	at metabase.mbql.normalize$fn__19352.invokeStatic(normalize.cljc:404)
	at metabase.mbql.normalize$fn__19352.invoke(normalize.cljc:398)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at metabase.mbql.normalize$canonicalize_implicit_field_id.invokeStatic(normalize.cljc:363)
	at metabase.mbql.normalize$canonicalize_implicit_field_id.invoke(normalize.cljc:360)
	at metabase.mbql.normalize$canonicalize_simple_filter_clause.invokeStatic(normalize.cljc:455)
	at metabase.mbql.normalize$canonicalize_simple_filter_clause.invoke(normalize.cljc:452)
	at metabase.mbql.normalize$fn__19409$fn__19410.invoke(normalize.cljc:464)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at metabase.mbql.normalize$canonicalize_mbql_clauses$fn__19524.invoke(normalize.cljc:552)
	... 218 more
2021-06-11 19:26:21,300 DEBUG middleware.log :: GET /api/card/654 200 172.5 ms (5 DB calls) App DB connections: 1/7 Jetty threads: 6/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
2021-06-11 19:26:21,358 DEBUG middleware.log :: GET /api/database 200 242.2 ms (24 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
2021-06-11 19:26:21,510 DEBUG middleware.log :: GET /api/database 200 395.4 ms (4 DB calls) App DB connections: 0/7 Jetty threads: 4/50 (3 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued)
```

</details>

Doing steps 1+2 on 0.39+ will just create a small query error with the possibility to modify/fix the question in the UI.
<details><summary>Error from doing everything on 0.39</summary>

```
2021-06-11 19:28:54,911 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 4,
 :started_at #t "2021-06-11T19:28:54.758431+02:00[Europe/Copenhagen]",
 :error_type :invalid-query,
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:js-int-to-string? true},
  :database 4,
  :query
  {:source-table "card__645",
   :filter [:< [:field "MYDATE" {:base-type :type/Date, :temporal-unit :minute}] "2021-06-11T12:30:00"]},
  :parameters [],
  :async? true,
  :cache-ttl nil},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> util.schema$schema_core_validator$fn__16373.invoke(schema.clj:29)"
  "query_processor.middleware.validate$validate_query$fn__49643.invoke(validate.clj:9)"
  "query_processor.middleware.normalize_query$normalize$fn__47524.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__44790.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49574.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__46641.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__48686.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46581.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___38051$thunk__38052.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___38051$fn__38054.invoke(reducible.clj:108)"],
 :context :question,
 :error
 "Value does not match schema: {:query {:filter (named [nil (named (named [nil nil (named (not (\"Invalid :temporal-unit for the specified :base-type.\" a-clojure.lang.PersistentArrayMap)) \"options\")] \"Must be a valid instance of one of these clauses: :expression, :field\") \"field\") nil] \"Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :starts-with, :ends-with, :contains, :does-not-contain, :inside, :is-empty, :not-empty, :is-null, :not-null, :time-interval, :segment\")}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :schema.core/error,
  :value
  {:constraints {:max-results 10000, :max-results-bare-rows 2000},
   :type :query,
   :middleware {:js-int-to-string? true},
   :info
   {:executed-by 1,
    :context :question,
    :card-id 646,
    :query-hash
    [120, 127, -1, -4, 76, -62, -31, 28, 87, -5, -84, 67, -121, 99, -117, 76, 54, 61, -39, -97, -125, 120, -34, -14,
     66, 79, -29, 81, -74, 42, -73, 43]},
   :database 4,
   :query
   {:source-table "card__645",
    :filter [:< [:field "MYDATE" {:base-type :type/Date, :temporal-unit :minute}] "2021-06-11T12:30:00"]},
   :async? true},
  :error
  {:query
   {:filter
    (named [nil (named (named [nil nil (named (not ("Invalid :temporal-unit for the specified :base-type." a-clojure.lang.PersistentArrayMap)) "options")] "Must be a valid instance of one of these clauses: :expression, :field") "field") nil] "Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :starts-with, :ends-with, :contains, :does-not-contain, :inside, :is-empty, :not-empty, :is-null, :not-null, :time-interval, :segment")}}},
 :data {:rows [], :cols []}}

2021-06-11 19:28:54,916 DEBUG middleware.log :: POST /api/card/646/query 202 [ASYNC: completed] 168.8 ms (3 DB calls) App DB connections: 0/7 Jetty threads: 3/50 (4 idle, 0 queued) (72 total active threads) Queries in flight: 1 (0 queued)
```

</details>

**Solution:**
1. Handling the conversion of old field dimensions during query processor
2. Removing the option to define time for date-only fields in the UI.

--- a/shared/src/metabase/mbql/normalize.cljc
+++ b/shared/src/metabase/mbql/normalize.cljc
@@ -365,7 +365,7 @@
         :else
         x)
       (catch #?(:clj Throwable :cljs js/Error) e
-        (throw (ex-info (i18n/tru "Error normalizing form.")
+        (throw (ex-info (i18n/tru "Error normalizing form: {0}" (ex-message e))
                         {:form x, :path path, :special-fn special-fn}
                         e))))))
 
@@ -589,7 +589,7 @@
          (canonicalize-mbql-clause x)
          (catch #?(:clj Throwable :cljs js/Error) e
            (log/error (i18n/tru "Invalid clause:") x)
-           (throw (ex-info (i18n/tru "Invalid MBQL clause")
+           (throw (ex-info (i18n/tru "Invalid MBQL clause: {0}" (ex-message e))
                            {:clause x}
                            e))))))
    mbql-query))

--- a/shared/src/metabase/mbql/schema.cljc
+++ b/shared/src/metabase/mbql/schema.cljc
@@ -293,21 +293,33 @@
       validate-bin-width
       validate-num-bins))
 
+(defn valid-temporal-unit-for-base-type?
+  "Whether `temporal-unit` (e.g. `:day`) is valid for the given `base-type` (e.g. `:type/Date`). If either is `nil` this
+  will return truthy. Accepts either map of `field-options` or `base-type` and `temporal-unit` passed separately."
+  ([{:keys [base-type temporal-unit] :as _field-options}]
+   (valid-temporal-unit-for-base-type? base-type temporal-unit))
+
+  ([base-type temporal-unit]
+   (cond
+     (core/not temporal-unit)
+     true
+
+     (core/not base-type)
+     true
+
+     :else
+     (let [units (condp #(isa? %2 %1) base-type
+                   :type/Date     date-bucketing-units
+                   :type/Time     time-bucketing-units
+                   :type/DateTime datetime-bucketing-units)]
+       (contains? units temporal-unit)))))
+
 (defn- validate-temporal-unit [schema]
   ;; TODO - consider breaking this out into separate constraints for the three different types so we can generate more
   ;; specific error messages
   (s/constrained
    schema
-   (fn [{:keys [base-type temporal-unit]}]
-     (if-not temporal-unit
-       true
-       (if-let [units (condp #(isa? %2 %1) base-type
-                        :type/Date     date-bucketing-units
-                        :type/Time     time-bucketing-units
-                        :type/DateTime datetime-bucketing-units
-                        nil)]
-         (contains? units temporal-unit)
-         true)))
+   valid-temporal-unit-for-base-type?
    "Invalid :temporal-unit for the specified :base-type."))
 
 (defn- no-binning-options-at-top-level [schema]

--- a/shared/src/metabase/mbql/schema.cljc
+++ b/shared/src/metabase/mbql/schema.cljc
@@ -300,19 +300,14 @@
    (valid-temporal-unit-for-base-type? base-type temporal-unit))
 
   ([base-type temporal-unit]
-   (cond
-     (core/not temporal-unit)
-     true
-
-     (core/not base-type)
-     true
-
-     :else
-     (let [units (condp #(isa? %2 %1) base-type
-                   :type/Date     date-bucketing-units
-                   :type/Time     time-bucketing-units
-                   :type/DateTime datetime-bucketing-units)]
-       (contains? units temporal-unit)))))
+   (if-let [units (when (core/and temporal-unit base-type)
+                    (condp #(isa? %2 %1) base-type
+                      :type/Date     date-bucketing-units
+                      :type/Time     time-bucketing-units
+                      :type/DateTime datetime-bucketing-units
+                      nil))]
+     (contains? units temporal-unit)
+     true)))
 
 (defn- validate-temporal-unit [schema]
   ;; TODO - consider breaking this out into separate constraints for the three different types so we can generate more

--- a/shared/src/metabase/mbql/util.cljc
+++ b/shared/src/metabase/mbql/util.cljc
@@ -4,10 +4,12 @@
   #?@
   (:clj
    [(:require [clojure.string :as str]
+              [clojure.tools.logging :as log]
               [metabase.mbql.schema :as mbql.s]
               [metabase.mbql.schema.helpers :as schema.helpers]
               [metabase.mbql.util.match :as mbql.match]
               [metabase.shared.util.i18n :as i18n]
+              metabase.util.i18n
               [potemkin :as p]
               [schema.core :as s])]
    :cljs
@@ -661,10 +663,17 @@
 
 (defn with-temporal-unit
   "Set the `:temporal-unit` of a `:field` clause to `unit`."
-  [clause unit]
+  [[_ _ {:keys [base-type]} :as clause] unit]
   ;; it doesn't make sense to call this on an `:expression` or `:aggregation`.
   (assert (is-clause? :field clause))
-  (assoc-field-options clause :temporal-unit unit))
+  (if (or (not base-type)
+          (mbql.s/valid-temporal-unit-for-base-type? base-type unit))
+    (assoc-field-options clause :temporal-unit unit)
+    (do
+      #?(:clj
+         (log/warn (metabase.util.i18n/trs "{0} is not a valid temporal unit for {1}; not adding to clause {2}"
+                                           unit base-type (pr-str clause))))
+      clause)))
 
 (defn remove-namespaced-options
   "Update a `:field`, `:expression` reference, or `:aggregation` reference clause by removing all namespaced keys in the

--- a/shared/test/metabase/mbql/normalize_test.cljc
+++ b/shared/test/metabase/mbql/normalize_test.cljc
@@ -419,7 +419,7 @@
     "Make sure token normalization works correctly on source queries"
     {{:database 4
       :type     :query
-      :query    {"source_query" {:native         "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10",
+      :query    {"source_query" {:native         "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
                                  "template_tags" {:category {:name         "category"
                                                              :display-name "Category"
                                                              :type         "text"
@@ -427,7 +427,7 @@
                                                              :default      "Widget"}}}}}
      {:database 4
       :type     :query
-      :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10",
+      :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
                                 :template-tags {"category" {:name         "category"
                                                             :display-name "Category"
                                                             :type         :text
@@ -437,7 +437,7 @@
      {:database 4
       :type     :query
       :query    {"source_query" {"source_table" 1, "aggregation" "rows"}}}
-     {:database 4,
+     {:database 4
       :type     :query
       :query    {:source-query {:source-table 1, :aggregation :rows}}}}))
 
@@ -558,7 +558,7 @@
 
    "expressions should handle datetime arithemtics"
    {{:query {:expressions {:prev_month ["+" ["field-id" 13] ["interval" -1 "month"]]}}}
-    {:query {:expressions {"prev_month" [:+ [:field-id 13] [:interval -1 :month]]}}},
+    {:query {:expressions {"prev_month" [:+ [:field-id 13] [:interval -1 :month]]}}}
 
     {:query {:expressions {:prev_month ["-" ["field-id" 13] ["interval" 1 "month"] ["interval" 1 "day"]]}}}
     {:query {:expressions {"prev_month" [:- [:field-id 13] [:interval 1 :month] [:interval 1 :day]]}}}}
@@ -854,7 +854,7 @@
       :query    {:filter [:and
                           [:segment "gaid:-11"]
                           [:time-interval [:field-id 6851] -365 :day {}]]}}
-     {:database 1,
+     {:database 1
       :type     :query
       :query    {:filter
                  [:and
@@ -943,7 +943,7 @@
                                                             :default      "Widget"}}}}}
      {:database 4
       :type     :query
-      :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10",
+      :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
                                 :template-tags {"category" {:name         "category"
                                                             :display-name "Category"
                                                             :type         :text
@@ -1122,7 +1122,7 @@
   (t/testing "make sure source queries get normalized properly!"
     (t/is (= {:database 4
               :type     :query
-              :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10",
+              :query    {:source-query {:native        "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
                                         :template-tags {"category" {:name         "category"
                                                                     :display-name "Category"
                                                                     :type         :text
@@ -1131,7 +1131,7 @@
              (mbql.normalize/normalize
               {:database 4
                :type     :query
-               :query    {"source_query" {:native         "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10",
+               :query    {"source_query" {:native         "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
                                           "template_tags" {:category {:name         "category"
                                                                       :display-name "Category"
                                                                       :type         "text"
@@ -1412,3 +1412,14 @@
                       :path       []
                       :special-fn nil}
                      (ex-data cause)))))))))
+
+(t/deftest ^:parallel remove-unsuitable-temporal-units-test
+  (t/testing "Ignore unsuitable temporal units (such as bucketing a Date by minute) rather than erroring (#16485)"
+    ;; this query is with legacy MBQL syntax. It's just copied directly from the original issue
+    (let [query {:query {:filter ["<"
+                                  ["datetime-field" ["field-literal" "date_seen" "type/Date"] "minute"]
+                                  "2021-05-01T12:30:00"]}}]
+      (t/is (= {:query {:filter [:<
+                                 [:field "date_seen" {:base-type :type/Date}]
+                                 "2021-05-01T12:30:00"]}}
+               (mbql.normalize/normalize query))))))

--- a/shared/test/metabase/mbql/normalize_test.cljc
+++ b/shared/test/metabase/mbql/normalize_test.cljc
@@ -1402,12 +1402,9 @@
           (t/is (= {:query bad-query}
                    (ex-data e))))
         (t/testing "\nParent exception(s) should be even more specific"
-          (let [cause #?(:clj (some-> ^Throwable e .getCause)
-                         :cljs (ex-cause e))]
+          (let [cause (ex-cause e)]
             (t/is (some? cause))
-            (t/is (= "Error normalizing form."
-                     #?(:clj (.getMessage cause)
-                        :cljs (ex-message cause))))
+            (t/is (re-find #"Error normalizing form:" (ex-message cause)))
             (t/is (= {:form       bad-query
                       :path       []
                       :special-fn nil}

--- a/shared/test/metabase/mbql/schema_test.cljc
+++ b/shared/test/metabase/mbql/schema_test.cljc
@@ -5,27 +5,28 @@
 
 (t/deftest ^:parallel field-clause-test
   (t/testing "Make sure our schema validates `:field` clauses correctly"
-    (t/are [clause expected] (= expected
-                                (not (s/check mbql.s/field clause)))
-      [:field 1 nil]                                                          true
-      [:field 1 {}]                                                           true
-      [:field 1 {:x true}]                                                    true
-      [:field 1 2]                                                            false
-      [:field "wow" nil]                                                      false
-      [:field "wow" {}]                                                       false
-      [:field "wow" 1]                                                        false
-      [:field "wow" {:base-type :type/Integer}]                               true
-      [:field "wow" {:base-type 100}]                                         false
-      [:field "wow" {:base-type :type/Integer, :temporal-unit :month}]        true
-      [:field "wow" {:base-type :type/Date, :temporal-unit :month}]           true
-      [:field "wow" {:base-type :type/DateTimeWithTZ, :temporal-unit :month}] true
-      [:field "wow" {:base-type :type/Time, :temporal-unit :month}]           false
-      [:field 1 {:binning {:strategy :num-bins}}]                             false
-      [:field 1 {:binning {:strategy :num-bins, :num-bins 1}}]                true
-      [:field 1 {:binning {:strategy :num-bins, :num-bins 1.5}}]              false
-      [:field 1 {:binning {:strategy :num-bins, :num-bins -1}}]               false
-      [:field 1 {:binning {:strategy :default}}]                              true
-      [:field 1 {:binning {:strategy :fake}}]                                 false)))
+    (doseq [[clause expected] {[:field 1 nil]                                                          true
+                               [:field 1 {}]                                                           true
+                               [:field 1 {:x true}]                                                    true
+                               [:field 1 2]                                                            false
+                               [:field "wow" nil]                                                      false
+                               [:field "wow" {}]                                                       false
+                               [:field "wow" 1]                                                        false
+                               [:field "wow" {:base-type :type/Integer}]                               true
+                               [:field "wow" {:base-type 100}]                                         false
+                               [:field "wow" {:base-type :type/Integer, :temporal-unit :month}]        true
+                               [:field "wow" {:base-type :type/Date, :temporal-unit :month}]           true
+                               [:field "wow" {:base-type :type/DateTimeWithTZ, :temporal-unit :month}] true
+                               [:field "wow" {:base-type :type/Time, :temporal-unit :month}]           false
+                               [:field 1 {:binning {:strategy :num-bins}}]                             false
+                               [:field 1 {:binning {:strategy :num-bins, :num-bins 1}}]                true
+                               [:field 1 {:binning {:strategy :num-bins, :num-bins 1.5}}]              false
+                               [:field 1 {:binning {:strategy :num-bins, :num-bins -1}}]               false
+                               [:field 1 {:binning {:strategy :default}}]                              true
+                               [:field 1 {:binning {:strategy :fake}}]                                 false}]
+      (t/testing (pr-str clause)
+        (t/is (= expected
+                 (not (s/check mbql.s/field clause))))))))
 
 (t/deftest ^:parallel validate-template-tag-names-test
   (t/testing "template tags with mismatched keys/`:names` in definition should be disallowed\n"

--- a/shared/test/metabase/mbql/util_test.cljc
+++ b/shared/test/metabase/mbql/util_test.cljc
@@ -792,3 +792,13 @@
     [:aggregation 0]                              [:aggregation 0]
     [:aggregation 0 {::namespaced true}]          [:aggregation 0]
     [:aggregation 0 {::namespaced true, :a 1}]    [:aggregation 0 {:a 1}]))
+
+(t/deftest with-temporal-unit-test
+  (t/is (= [:field 1 {:temporal-unit :day}]
+           (mbql.u/with-temporal-unit [:field 1 nil] :day)))
+  (t/is (= [:field "t" {:base-type :type/Date, :temporal-unit :day}]
+           (mbql.u/with-temporal-unit [:field "t" {:base-type :type/Date}] :day)))
+  (t/testing "Ignore invalid temporal units if `:base-type` is specified (#16485)"
+    ;; `:minute` doesn't make sense for a DATE
+    (t/is (= [:field "t" {:base-type :type/Date}]
+             (mbql.u/with-temporal-unit [:field "t" {:base-type :type/Date}] :minute)))))

